### ISSUE TITLE
Xcode 16 "folder style" projects

### ISSUE THREAD

**Issue by mickeyl** [tuist/XcodeProj#880]
With Xcode 16, Apple has set the folder style (instead of group style) as the default for the Xcode project file organisation.

With these kinds of projects, XcodeProj no longer seems to find any resource files.

**Comment by AliTavakoliFFW** [comment]
I face the same problem, our project is using now the FolderStructure with xcode 16, therefore I think, it can't be read by XcodeProj.

**Comment by dusi** [comment]
We have recently experienced similar issue which makes it seem like XcodeProj does not support `FileSystemSynchronizedGroup` on the workspace scope 🤔 

For us this has [manifested](https://github.com/peripheryapp/periphery/issues/828#issuecomment-2571435324) by attempting to use [periphery](https://github.com/peripheryapp/periphery) which has a dependency on tuist's `XcodeProj`.

I'm attaching the same sample project just in case there's an "easy" fix for this.
[Sample.zip](https://github.com/user-attachments/files/18323874/Sample.zip)

**Comment by lukecharman** [comment]
Looks like we have a PR to fix it sitting behind a small linting error – hoping we can see this merged and released soon! :)

**Comment by fortmarek** [comment]
Yes, we have an automated release process, so what has been merged, has also been released.

Correct me if I'm wrong, but I believe we can close this one.

**Comment by AliTavakoliFFW** [comment]
I tried the new Version 8.27.3 and I still get nothing from sourceFiles() and resourcesBuildPhase()?.files(), no files.

**Comment by mickeyl** [comment]
Yeah, I'm seeing the same behavior here. @fortmarek Can we please reopen the issue?

**Comment by kwridan** [comment]
This appears to be a quirk of the Xcode project format as opposed to a bug in XcodeProj. 

Inspecting the raw contents of the pbxproj file when buildable folders (sync groups) are used, there doesn't appear to be any references to those folder in the sources build phase 🤔

```
/* Begin PBXSourcesBuildPhase section */
		FB17F7312D82E1C5005A08D0 /* Sources */ = {
			isa = PBXSourcesBuildPhase;
			buildActionMask = 2147483647;
			files = (
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXSourcesBuildPhase section */
```

**Comment by fortmarek** [comment]
> Inspecting the raw contents of the pbxproj file when buildable folders (sync groups) are used, there doesn't appear to be any references to those folder in the sources build phase 🤔

Yeah, the files are not represented in the `pbxproj` since that's dynamic and resolved by Xcode based on the synchronized group. To expand the files is not a responsibility of `XcodeProj` ... so I'm closing this again 😅

**Comment by mickeyl** [comment]
Ok, good for `XcodeProj`, bad for clients though. I wonder whether `XcodeProj` at least could assist these cases, i.e. with API that tells you that there is a folder (where?) we have to traverse manually.

**Comment by fortmarek** [comment]
> I wonder whether XcodeProj at least could assist these cases, i.e. with API that tells you that there is a folder (where?) we have to traverse manually.

You need to get the synchronized groups and expand the sources. We already do this in `XcodeGraph`: https://github.com/tuist/XcodeGraph/blob/bcbef4b34953a594adec24c472c24a2a4c6fce25/Sources/XcodeGraphMapper/Mappers/Targets/PBXTargetMapper.swift#L369-L407

fwiw, `XcodeProj` is a low-level library that's meant to be 1-1 mapping of the `pbxproj` format. For higher-level modifications, `XcodeGraph` might be a library you might want to look into, depending on what you're doing.

**Comment by AliTavakoliFFW** [comment]
@fortmarek Thanks for the response, I'm gonna check your alternative suggestion.
